pkg/apis/kops/validation/legacy.go

```diff
@@ -602,6 +602,23 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 
 	allErrs = append(allErrs, newValidateCluster(c)...)
 
+	if c.Spec.Networking != nil && c.Spec.Networking.Cilium != nil {
+		ciliumSpec := c.Spec.Networking.Cilium
+		if ciliumSpec.Ipam == "eni" {
+			if c.Spec.CloudProvider != "aws" {
+				allErrs = append(allErrs, field.Invalid(fieldSpec.Child("Cilium").Child("Ipam"), "eni", "Cilum ENI IPAM is supported only in AWS"))
+			}
+			if !ciliumSpec.DisableMasquerade {
+				allErrs = append(allErrs, field.Invalid(fieldSpec.Child("Cilium").Child("DisableMasquerade"), "false", "Masquerade must be disabled when ENI IPAM is used"))
```

Comment:
Also `field.Forbidden()`

Is masquerading forbidden with ENI? the Note block at https://docs.cilium.io/en/v1.6/gettingstarted/aws-eni/#prepare-deploy-cilium seems to suggest either way is possible.

Reply:
If I masq, cluster networking doesn't work. I am guessing Cilium is masking behind the wrong interface or something. I couldn't quite figure out why. It may be that one have to explicitly set which interface to mask behind. But then I am unsure if we can really know that across all distros.

Masqing only makes sense if the SG on the ENI cilium creates have egress rules. But there isn't really any benefit of doing that since because of the masq, traffic through the ENI reaches internet anyway.

So I think the easiest is to force masqing off for now and if anyone has any good reason for having masq on, we loosen that restriction.

Reply:
In ENI mode, there will be multiple possible egress paths for traffic leaving a node. Based on the destination, the right egress interface will be chosen. Masquerading is typically only ever needed for traffic heading out to the interface. This is typically routed via the default route pointing out `eth0` although there can be node configurations where this is different.

For this reason, the Cilium ENI guide sets the `egressMasqueradingInterfaces` helm option to limit masquerading to traffic leaving on eth0:
```
  --set global.egressMasqueradeInterfaces=eth0 \
```

Masquerading is *only* required if no NAT/internet gateway is set up in AWS which will perform the masquerading.